kernel/arch/i386/interrupt.S

```diff
@@ -0,0 +1,390 @@
+    push 0
+    push 5
+    jmp isr_common_stub
+
+; 6: Invalid Opcode Exception
+global isr6
+isr6:
+    push 0
+    push 6
+    jmp isr_common_stub
+
+; 7: Coprocessor Not Available Exception
+global isr7
+isr7:
+    push 0
+    push 7
+    jmp isr_common_stub
+
+; 8: Double Fault Exception (With Error Code!)
+global isr8
+isr8:
+    push 8
+    jmp isr_common_stub
+
+; 9: Coprocessor Segment Overrun Exception
+global isr9
+isr9:
+    push 0
+    push 9
+    jmp isr_common_stub
+
+; 10: Bad TSS Exception (With Error Code!)
+global isr10
+isr10:
+    push 10
+    jmp isr_common_stub
+
+; 11: Segment Not Present Exception (With Error Code!)
+global isr11
+isr11:
+    push 11
+    jmp isr_common_stub
+
+; 12: Stack Fault Exception (With Error Code!)
+global isr12
+isr12:
+    push 12
+    jmp isr_common_stub
+
+; 13: General Protection Fault Exception (With Error Code!)
+global isr13
+isr13:
+    push 13
+    jmp isr_common_stub
+
+; 14: Page Fault Exception (With Error Code!)
+global isr14
+isr14:
+    push 14
+    jmp isr_common_stub
+
+; 15: Reserved Exception
+global isr15
+isr15:
+    push 0
+    push 15
+    jmp isr_common_stub
+
+; 16: Floating Point Exception
+global isr16
+isr16:
+    push 0
+    push 16
+    jmp isr_common_stub
+
+; 17: Alignment Check Exception (With Error Code!)
+global isr17
+isr17:
+    push 17
+    jmp isr_common_stub
+
+; 18: Machine Check Exception
+global isr18
+isr18:
+    push 0
+    push 18
+    jmp isr_common_stub
+
+; 19: Reserved
+global isr19
+isr19:
+    push 0
+    push 19
+    jmp isr_common_stub
+
+; 20: Reserved
+global isr20
+isr20:
+    push 0
+    push 20
+    jmp isr_common_stub
+
+; 21: Reserved
+global isr21
+isr21:
+    push 0
+    push 21
+    jmp isr_common_stub
+
+; 22: Reserved
+global isr22
+isr22:
+    push 0
+    push 22
+    jmp isr_common_stub
+
+; 23: Reserved
+global isr23
+isr23:
+    push 0
+    push 23
+    jmp isr_common_stub
+
+; 24: Reserved
+global isr24
+isr24:
+    push 0
+    push 24
+    jmp isr_common_stub
+
+; 25: Reserved
+global isr25
+isr25:
+    push 0
+    push 25
+    jmp isr_common_stub
+
+; 26: Reserved
+global isr26
+isr26:
+    push 0
+    push 26
+    jmp isr_common_stub
+
+; 27: Reserved
+global isr27
+isr27:
+    push 0
+    push 27
+    jmp isr_common_stub
+
+; 28: Reserved
+global isr28
+isr28:
+    push 0
+    push 28
+    jmp isr_common_stub
+
+; 29: Reserved
+global isr29
+isr29:
+    push 0
+    push 29
+    jmp isr_common_stub
+
+; 30: Reserved (With Error Code!)
+global isr30
+isr30:
+    push 30
+    jmp isr_common_stub
+
+; 31: Reserved
+global isr31
+isr31:
+    push 0
+    push 31
+    jmp isr_common_stub
+
+; All of the following Interrupt Requests are
+; stubs for hardware interrupts such as device interrupts.
+; These are not exceptions. Each Interrupt Request pushes
+; the IRQ value (starting at 0) along with their
+; corresponding hardware interrupt value (starting at 32).
+;
+; Interrupt Request handlers
+; System timer (PIT)
+global irq0
```

Comment:
Same thing as question about the ISR, but with modifications for IRQs.

Reply:
Yes, both of them would be included.  Same as above.